app/models/team.rb

```diff
@@ -45,18 +45,23 @@ def visible
     end
 
     # Returns a base scope reflecting the relevant teams depending on what
-    # phase of the running season we're currently in.
+    # phase of the running season we're currentin_ly in.
     def by_season_phase
       if Time.now.utc > Season.current.acceptance_notification_at
-        Team.where season: Season.current, kind: %w(sponsored voluntary)
+        Team.in_current_season.selected
       else
-        Team.visible.where season: Season.current
+        Team.in_current_season.visible
       end
     end
 
-    def active_teams
-      Team.where(season: Season.current, kind: %w(sponsored voluntary)).order(:name)
+    def in_current_season
```

Comment:
I like it very much! And thanks A LOT for the explanation. 

For the follow-up PR: I checked the factories, I didn't find a factory that uses the season scope of the Team class. The current_season traits all call the class method Season.current. 
You don't mean that the factory trait should have the Team scope instead, do you? 
And we are talking only about renaming current_season to in_current_season? Not current_other_things. 
 
If I understand it well: just renaming 'current_season' scope in Team class, to in_current_season scope: that is a tiny job, I can do it right away.
Otherwise, please merge and I will open a new PR and figure it out.